Core/FileStore.swift

```diff
@@ -22,16 +22,30 @@ import Configuration
 
 public class FileStore {
 
-    private let groupIdentifier: String = ContentBlockerStoreConstants.groupName
+    private let groupIdentifier: String = ContentBlockerStoreConstants.configurationGroupName
 
     public init() { }
 
     public func persist(_ data: Data, for configuration: Configuration) throws {
-        do {
-            try data.write(to: persistenceLocation(for: configuration), options: .atomic)
-        } catch {
-            Pixel.fire(pixel: .fileStoreWriteFailed, error: error, withAdditionalParameters: ["config": configuration.rawValue])
-            throw error
+        let file = persistenceLocation(for: configuration)
+        var coordinatorError: NSError?
+        var writeError: Error?
+
+        NSFileCoordinator().coordinate(writingItemAt: file, options: .forReplacing, error: &coordinatorError) { fileUrl in
```

Comment:
Let's check that this `persist` function overall is never called on the main thread, since the `coordinate` function will block that thread until it can complete what it's trying to do.

Reply:
I tracked the persist calls all the way up to [here](https://github.com/duckduckgo/iOS/blob/1c736e9c4d9befca0a42ed280b4d47c2a5f38800/DuckDuckGo/AppConfigurationFetch.swift#L172). Basically the full fetch then store process is initiated by the `AppConfigurationFetch` within a `Task`. AFAIK that should put it off the main thread.